src/coreneuron/io/reports/report_event.cpp

```diff
@@ -72,30 +69,20 @@ void ReportEvent::summation_alu(NrnThread* nt) {
     }
 }
 
-/** on deliver, call ReportingLib and setup next event */
+/** on deliver, call libsonata and setup next event */
 void ReportEvent::deliver(double t, NetCvode* nc, NrnThread* nt) {
-/* reportinglib is not thread safe */
-#pragma omp critical
-    {
```

Comment:
The code comment was bit misleading I believe. In the same way, libsonatareport is not also thread safe and hence I believe we still need to keep `omp critical` here.